database/factories/BookFactory.php

```diff
@@ -1,39 +1,34 @@
 <?php
 
+use App\Book;
+use App\User;
+use App\School;
+use App\Myclass;
 use Faker\Generator as Faker;
 
-$factory->define(App\Book::class, function (Faker $faker) {
+$factory->define(Book::class, function (Faker $faker) {
     return [
         'book_code' => 'bk'.$faker->unique()->randomNumber(7, false),
-        'title' => $faker->sentences(1, true),
-        'author' => $faker->name,
-        'quantity' => $faker->randomElement([5,8,19,13,34]),
-        'rackNo' => $faker->randomElement([1,2,3,4,5,6,7,8,9,10,11,12]),
-        'rowNo' => $faker->randomElement([1,2,3,4,5,6,7,8,9,10,11,12]),
-        'type' => $faker->randomElement(['Academic','Magazine','Story','Other']),
-        'img_path' => $faker->url,
-        'about' => $faker->sentences(3, true),
-        'price' => $faker->randomNumber,
-        'class_id' => function () use ($faker) {
-          if(App\Myclass::count() == 0)
-            return factory(App\Myclass::class)->create()->id;
-          else {
-            return $faker->randomElement(App\Myclass::pluck('id')->toArray());
-          }
-        },
-        'school_id' => function () use ($faker) {
-          if(App\School::count() == 0)
-            return factory(App\School::class)->create()->id;
-          else {
-            return $faker->randomElement(App\School::pluck('id')->toArray());
-          }
-        },
-        'user_id' => function () use ($faker) {
-          if(App\User::count() == 0)
-            return factory(App\User::class)->create()->id;
-          else {
-            return $faker->randomElement(App\User::pluck('id')->toArray());
-          }
-        },
+        'title'     => $faker->sentences(1, true),
+        'author'    => $faker->name,
+        'quantity'  => $faker->randomElement([5,8,19,13,34]),
+        'rackNo'    => $faker->randomElement([1,2,3,4,5,6,7,8,9,10,11,12]),
+        'rowNo'     => $faker->randomElement([1,2,3,4,5,6,7,8,9,10,11,12]),
+        'type'      => $faker->randomElement(['Academic','Magazine','Story','Other']),
+        'img_path'  => $faker->imageUrl($width = 150, $height = 150, 'cats'),
+        'about'     => $faker->sentences(3, true),
+        'price'     => $faker->randomNumber,
+        'class_id'  => function() {
```

Comment:
I used Faker random to get random value. If you only use first(), then you will always get first value and it will remain same since e.g.: `Myclass::count()` returns always true if there is even a single row. Let me know what do you think.

Reply:
@changeweb I needed to use ?
``` php
if (Myclass::count()) $faker->randomElement(App\User::pluck('id')->toArray());
else return factory(Myclass::class)->create()->id;
```

Reply:
`App\User` would be `Myclass`.
And do you get my point about random value instead of a fixed one? Let me know what do you think about using random value instead of a fixed one for all rows.

I think using random values will give more variation in data.

Reply:
@changeweb you are right!
I changed it